Add More TextSplitter Options

iam-vipin opened the pull request:
#20 

**1 Recursive**  :  
1. from_language
2. by_character

**2 Character** 

**3 Token**

**4 Header** : 

1. Markdown
2. HTML

### Video
[Screencast from 2023-11-13 14-56-26.webm](https://github.com/llmshare/llmproto/assets/99081689/e8c6bf6b-e192-4858-85f1-674ccfbd9600)
### Image
![image](https://github.com/llmshare/llmproto/assets/99081689/6160b1d2-ef8a-4596-9cf3-bc0361356f29)

iam-vipin committed:
feat: add CharacterTextSplitter

iam-vipin commented:
Hey @Sing-Li  I added one new TextSplitter can you please Review it so I can implement other using the same way.
Cause right now we did not have any way to verfiy there  working behind the scene but i made sure it look and  work's perfectly fine.

Sing-Li commented:
Looks good @VipinDevelops       

Advise to *add more arguments that are frequently used*, for example in this case:

`separator, chunk_size, chunk_overlap,  length_function,  is_separator_regex`

See usage in sample...

https://github.com/langchain-ai/langchain/blob/31f264169db4ab23689f2e179983f1cfdfd1a33a/docs/docs/modules/data_connection/document_transformers/text_splitters/character_text_splitter.mdx#L17

Otherwise, yes, pls duplicate this for the rest of the blocks.

iam-vipin committed:
feat(text splitter): add TokenTextSplitter

iam-vipin commented:
> Looks good @VipinDevelops
> 
> Advise to _add more arguments that are frequently used_, for example in this case:
> 
> `separator, chunk_size, chunk_overlap, length_function, is_separator_regex`
> 
> See usage in sample...
> 
> https://github.com/langchain-ai/langchain/blob/31f264169db4ab23689f2e179983f1cfdfd1a33a/docs/docs/modules/data_connection/document_transformers/text_splitters/character_text_splitter.mdx#L17
> 
> Otherwise, yes, pls duplicate this for the rest of the blocks.

I added One more Now we have total 3 Type of TextSplitter do we need more or 3 are enough ? 
and Yeah I surely Add more arugments as well to all the TextSplitter.

iam-vipin committed:
feat: add chunksize, separator parameter

iam-vipin committed:
fix: fix size of splitter nodes

iam-vipin commented:
Hey @Sing-Li I added more arguments like Separator, chunkoverlap  to TextSplitter Node 
![image](https://github.com/llmshare/llmproto/assets/99081689/f26733e3-fa23-4782-88a8-cd7ffa0d4777)

iam-vipin commented:
Hey @Sing-Li Can you please let me know,
If Is there anything else needed to be done in this PR or its good to go ?

Sing-Li commented:
@VipinDevelops Can you please read this? https://github.com/llmshare/llmproto/issues/20

Please implement the other classes mentioned as well. Also please check actual usage of the class in code - and implement those frequently used arguments .. not all classes have the same arguments.

iam-vipin commented:
[Recursive](https://github.com/langchain-ai/langchain/blob/31f264169db4ab23689f2e179983f1cfdfd1a33a/docs/docs/modules/data_connection/document_transformers/text_splitters/recursive_text_splitter.mdx#L21)
![image](https://github.com/llmshare/llmproto/assets/99081689/1743fefe-7cc2-4580-b0b3-2aff1d28bcce)
![image](https://github.com/llmshare/llmproto/assets/99081689/bb419fa0-8787-49bf-b4ed-bd0149c978ac)

[Character](https://github.com/langchain-ai/langchain/blob/31f264169db4ab23689f2e179983f1cfdfd1a33a/docs/docs/modules/data_connection/document_transformers/text_splitters/character_text_splitter.mdx#L17)
![image](https://github.com/llmshare/llmproto/assets/99081689/f72c94dd-bdc7-40be-ac36-b2c26f92b331)
![image](https://github.com/llmshare/llmproto/assets/99081689/5a481b0d-decd-46ae-a247-6e64530a85fd)

I didn't added the `length_function` because its a function and I guess we don't assume to pass function in input of a block.

I only found the use of these two in the [folder](https://github.com/langchain-ai/langchain/tree/31f264169db4ab23689f2e179983f1cfdfd1a33a/docs/docs/modules/data_connection/document_transformers/text_splitters) you shared above.

https://github.com/langchain-ai/langchainjs/tree/main/examples/src/indexes

Sing-Li commented:
@VipinDevelops  pls look in the entire repository for coding examples, not just the "folder"

Also -pls implement all the other TextSplitters, including MarkdownHeaderTextSplitter

iam-vipin committed:
feat: update charactertextsplitter

iam-vipin committed:
feat: finish recursivechar textsplitter

iam-vipin committed:
feat: complete tokentextsplitter

iam-vipin committed:
feat: updated route handler char and recursive splitters

iam-vipin committed:
feat: setup token splitter routes

iam-vipin committed:
feat: added markdown header splitter

iam-vipin commented:
Hey, @Sing-Li  I've updated all the text splitter components, and they are ready for your review. I encountered some challenges when trying to identify the most commonly used splitters.so I went Langchain's official documentation and discovered that they have distinct documentation for both Python and JavaScript. Notably, certain splitters are exclusively available in Python, while others are limited to JavaScript. 

So dived deep how they all are related also read alot of code for that few of them are just part of each other and i updated  menu.

### JS/TS

- [Docs](https://js.langchain.com/docs/modules/data_connection/document_transformers/text_splitters/token )
- [code](https://github.com/langchain-ai/langchainjs/blob/main/langchain/src/text_splitter.ts)
![image](https://github.com/llmshare/llmproto/assets/99081689/a3b0ab1e-d90a-47eb-9c56-1d7bcd63fa5b)

### Python 

- [Docs](https://python.langchain.com/docs/modules/data_connection/document_transformers/text_splitters/split_by_token)  
- [Code](https://github.com/langchain-ai/langchain/blob/b2138508cbf17aef3a08cdce072d79fd5dc63dc3/libs/langchain/langchain/text_splitter.py)
![image](https://github.com/llmshare/llmproto/assets/99081689/077e1736-7fbb-42dd-9459-bc0d8834221b)

### What we have now 
![image](https://github.com/llmshare/llmproto/assets/99081689/9bb975d4-8d2a-4e12-866f-2fa801fc442e)

**Common** means that the text splitter is available in both Python and JS.

**JS** is specifically for the JavaScript Langchain.

**Python** is exclusively for the Python Langchain.

iam-vipin committed:
feat: added header splitter

Sing-Li commented:
@VipinDevelops what about `LatexTextSplitter`  as per spec?

iam-vipin committed:
docs: added removed lines

iam-vipin committed:
feat: add language text splitter

iam-vipin committed:
Merge branch 'main' into code-gen-blocks

iam-vipin commented:
Hey, @Sing-Li  I've updated all the text splitter components.
